src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-server.cpp

```diff
@@ -0,0 +1,369 @@
+
+        break;
+
+    case Commands::ModifyPushTransport::Id:
+        ChipLogDetail(Zcl, "PushAVStreamTransport: Modifying Push Transport");
+
+        HandleCommand<Commands::ModifyPushTransport::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandleModifyPushTransport(ctx, commandData); });
+
+        break;
+
+    case Commands::SetTransportStatus::Id:
+        ChipLogDetail(Zcl, "PushAVStreamTransport: Setting Push Transport Status");
+
+        HandleCommand<Commands::SetTransportStatus::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleSetTransportStatus(ctx, commandData); });
+
+        break;
+
+    case Commands::ManuallyTriggerTransport::Id:
+        ChipLogDetail(Zcl, "PushAVStreamTransport: Manually Triggered Push Transport");
+
+        HandleCommand<Commands::ManuallyTriggerTransport::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandleManuallyTriggerTransport(ctx, commandData); });
+
+        break;
+
+    case Commands::FindTransport::Id:
+        ChipLogDetail(Zcl, "PushAVStreamTransport: Finding Push Transport");
+
+        HandleCommand<Commands::FindTransport::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleFindTransport(ctx, commandData); });
+
+        break;
+    }
+}
+
+bool PushAvStreamTransportServer::FindStreamTransportConnection(const uint16_t connectionID)
+{
+    for (auto & id : mCurrentConnections)
+    {
+        if (id == connectionID)
+            return true;
+    }
+    return false;
+}
+
+uint16_t PushAvStreamTransportServer::GenerateConnectionID()
+{
+    static uint16_t assignedConnectionID = 0;
+    uint16_t nextConnectionID;
+
+    if (assignedConnectionID == MAX_PUSH_TRANSPORT_CONNECTION_ID)
+        nextConnectionID = 0;
+    else
+        nextConnectionID = assignedConnectionID + 1;
+
+    while (FindStreamTransportConnection(nextConnectionID) != false)
+    {
+        if (nextConnectionID == MAX_PUSH_TRANSPORT_CONNECTION_ID)
+            nextConnectionID = 0;
+        else
+            nextConnectionID = nextConnectionID + 1;
+    }
+    return nextConnectionID;
+}
+
+void PushAvStreamTransportServer::HandleAllocatePushTransport(HandlerContext & ctx,
+                                                              const Commands::AllocatePushTransport::DecodableType & commandData)
+{
+    Status status = Status::Success;
+    Commands::AllocatePushTransportResponse::Type response;
+    auto & transportOptions = commandData.transportOptions;
+
+    uint16_t connectionID                 = GenerateConnectionID();
+    TransportStatusEnum outTranportStatus = TransportStatusEnum::kUnknownEnumValue;
+
+    // call the delegate
+    status = mDelegate.AllocatePushTransport(connectionID, transportOptions, outTranportStatus);
```

Comment:
Where is the spec-required constraint validation of the fields of transportOptions?

That needs to happen here.